core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java

```diff
@@ -1366,18 +1412,10 @@ public static void removeTempFolder(Map<String, FolderDetails> locationMap, Stri
   /**
    * This method returns the list of index/merge index files for a segment in carbonTable.
    */
-  public static Set<String> getIndexFilesListForSegment(Segment segment, String tablePath)
-      throws IOException {
+  public static Set<String> getIndexFilesListForSegment(Segment segment, String tablePath) {
     Set<String> indexFiles;
-    if (segment.getSegmentFileName() == null) {
-      String segmentPath = CarbonTablePath.getSegmentPath(tablePath, segment.getSegmentNo());
-      indexFiles =
-          new SegmentIndexFileStore().getMergeOrIndexFilesFromSegment(segmentPath).keySet();
-    } else {
-      SegmentFileStore segmentFileStore =
```

Comment:
This method is used in case of SI data file merge to collect old and invalid files for segment. If it gets from segment file, we only get mergeindex file from previous load. So to collect index files also, we can get from the segment store.